sites/avivator/src/components/Controller/components/ChannelController.jsx

```diff
@@ -61,16 +64,37 @@ function ChannelController({
     shallow
   );
   const rgbColor = toRgb(colormap, color);
-  const [min, max] = domain;
-  // If the min/max range is and the dtype is float, make the step size smaller so contrastLimits are smoother.
+  const domainReducer = useCallback(
+    (state, action) => {
+      switch (action) {
+        case 'max/min': {
+          return domain;
+        }
+        case 'full': {
+          const { dtype } = loader[0];
+          const { max } = DTYPE_VALUES[dtype];
+          // Min is 0 for unsigned, or the negative of the max for signed dtypes.
+          const min =
+            dtype.startsWith('Int') || dtype.startsWith('Float') ? -max : 0;
+          return [min, max];
+        }
+        default: {
+          throw new Error();
+        }
+      }
+    },
+    [loader, domain]
+  );
+  const [[left, right], setDomain] = useReducer(domainReducer, domain);
```

Comment:
This is really difficult for me to reason with what's going on here. Why do we need a reducer? The `state` is never accessed. This feels like overkill for something that could be implemented with `useState` and a function.

```javascript
  const [mode, setMode] = useState("max/min");
  const [left, right] = getMinMax({ domain, mode, loader });
```

Reply:
There is an overhead to using `useCallback` or `useMemo` and we aren't doing anything expensive in determining the min max.

Reply:
Good observation, I can't remember having a good reason for doing it this way.